telephony/android_modem.c

```diff
@@ -2152,15 +2152,23 @@ handleNetworkRegistration( const char*  cmd, AModem  modem )
         cmd += 5;
         if (cmd[0] == '?') {
             if (modem->voice_mode == A_REGISTRATION_UNSOL_ENABLED_FULL) {
-                amodem_reply( modem, "+CREG: %d,%d, \"%04x\", \"%07x\"",
-                              modem->voice_mode, modem->voice_state,
-                              modem->area_code & 0xffff, modem->cell_id & 0xfffffff );
-                return;
-            }
-            else {
-                amodem_reply( modem, "+CREG: %d,%d",
-                              modem->voice_mode, modem->voice_state );
-                return;
+                // qemu/reference-ril does not follow standard format of network
+                // type in CGREG command, it uses the RIL_RadioTechnology in
+                // ril.h directly. To make it consistent, we use the same way in
+                // CREG, and also reuse the A_DATA_NETWORK_*.
```

Comment:
Ideally, we should follow the sepc as much as possible. However there is no spec for CDMA AT command at all. Considering the supporting of the CDMA, I am ok with this, but we still have to try to find a better solution. Thank you.